packages/noodl-viewer-react/src/nodes/navigation/navigate.js

```diff
@@ -174,33 +177,39 @@ function setup(context, graphModel) {
         if (Transitions[transition]) ports = ports.concat(Transitions[transition].ports(node.parameters));
       }
 
-      //    if(node.parameters['stack'] !== undefined) {
-      var pageStacks = graphModel.getNodesWithType('Page Stack');
-      var pageStack = pageStacks.find(
+      const pageStacks = graphModel.getNodesWithType('Page Stack');
+      const pageStack = pageStacks.find(
         (ps) => (ps.parameters['name'] || 'Main') === (node.parameters['stack'] || 'Main')
       );
 
       if (pageStack !== undefined) {
-        var pages = pageStack.parameters['pages'];
+        const pages = pageStack.parameters['pages'];
         if (pages !== undefined && pages.length > 0) {
           ports.push({
             plug: 'input',
-            type: { name: 'enum', enums: pages.map((p) => ({ label: p.label, value: p.id })), allowEditOnly: true },
+            type: {
+              name: 'enum',
+              enums: pages.map((p) => ({
+                label: p.label,
+                value: p.id
```

Comment:
The ids for the pages are unique and not exposed in the UI, this makes it a lot more challenging to allow the Target Page to be dynamically controlled via any input.